### PR TITLE
Add Pull Request Workflow

### PR DESCRIPTION
**Pull request by calebofearth**
Pull Request workflow is used to initiate all other workflows that run for a Pull Request
This includes a new pre-run workflow that checks a hash/timestamp value in the repository to verify code quality
(Integrators should run an internal workflow with proprietary tools/licenses, then update the timestamp/hash upon successful completion).
Depends upon #483 to complete (with registers regenerated using peakrdl updates)

### CODE REVIEW

**Review by bharatpillilli — DISMISSED** [review]

**Review by bharatpillilli — DISMISSED** [review]

File: .github/workflow_metadata/README.md
Comment (anjpar) [inline .github/workflow_metadata/README.md:1]: Nit pick. All bullets are numbered '1' :)

**Review by anjpar — CHANGES_REQUESTED** [review]
README.md needs to be updated with correct numbering for the numbered list.

**Review by anjpar — APPROVED** [review]

File: .github/workflows/pull_request.yml
Comment (korran) [inline .github/workflows/pull_request.yml:16]: 1. Why do we want to prevent the verilator tests from running if they don't have the magic hash/timestamp? Don't we want to give a drive-by contributors an early warning  if their modification doesn't even pass the verilator tests? Is this pre-check actually necessary?

File: .github/workflows/pull_request.yml
Comment (korran) [inline .github/workflows/pull_request.yml:22]: Again, I don't see why the pre-check needs to gate this step. It's useful to know if the RDL doesn't parse sooner than later.

File: .github/workflows/pull_request.yml
Comment (calebofearth) [reply .github/workflows/pull_request.yml:16]: Fair analysis - there's no specific reason to gate Verilator on the pre-run check, I can make them all run from time 0.
Validating the hash/timestamp is just a way to give feedback to make sure internal pipelines are run on the code before it gets merged. Keeping honest folks honest, if you will, and giving a record of internal sign-off.

Comment (korran) [reply .github/workflows/pull_request.yml]: Yeah, I understand the larger reason for having the check block merging, I just don't think it needs to be a pre-check for these other checks.

File: .github/workflows/pull_request.yml
Comment (calebofearth) [reply .github/workflows/pull_request.yml]: Pushed a commit to apply this feedback.

**Review by bharatpillilli — APPROVED** [review]

**Review by mcockrell-google — APPROVED** [review]

**Review by bharatpillilli — APPROVED** [review]

**Pull request merged**
